[Experimental Build] Merge tuinity

Spottedleaf opened the pull request:
This is an experimental build. This means you should be careful when using it, as it contains changes that are potentially dangerous. Only play around with it if you are OK with the risks.

This PR contains all of Tuinity's patches. Very notable ones are:
 - Highly optimised collisions
 - Optimised entity lookups by bounding box (Mojang made regressions in 1.17, this brings it back to 1.16)
 - Starlight https://github.com/PaperMC/Starlight
 - Rewritten dataconverter system https://github.com/PaperMC/DataConverter
 - Random block ticking optimisation (wrongly dropped from Paper 1.17)
 - Chunk ticking optimisations
 - Anything else I've forgotten in the 60 or so patches

If you are a previous Tuinity user, your config _will not_ migrate. You must do it yourself. The config options have simply been moved into paper.yml, so it will be an easy migration.

Please report any issues you have with this build to this thread only. 

This is the paperclip jar with these changes:
https://cdn.discordapp.com/attachments/876902758366736457/882145779735728138/Paper-1.17.1-R0.1-SNAPSHOT.jar
I will update it when needed, and write a reply when it's updated.

Spottedleaf committed:
Port tuinity, initial patchset

Spottedleaf committed:
Update gradle to 7.2

jmp said it fixes rebuildpatches not working for me. it fucking better

Spottedleaf committed:
Completely clean apply

Spottedleaf committed:
Remove tuinity config, add per player api patch

Spottedleaf committed:
Remove paper reobf mappings patch

Proximyst commented on patches/api/0329-Per-player-viewdistances.patch:
```diff
@@ -0,0 +1,112 @@
+      *
+      * @param viewDistance the player's view distance
+-     * @deprecated This is unimplemented and <i>will</i> throw an exception at runtime. The {@link org.bukkit.World World}-based methods still work.
++     * // Paper - implemented
```
How does this get rendered?

qixils replied:
pretty sure that should be `<!-- Paper - implemented -->`, otherwise it'll render in the javadocs

Proximyst commented on patches/server/0747-MC-Dev-fixes.patch:
```diff
@@ -0,0 +1,32 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Spottedleaf <Spottedleaf@users.noreply.github.com>
+Date: Wed, 16 Jun 2021 17:38:03 -0700
+Subject: [PATCH] MC-Dev fixes
```
Fixup

Proximyst commented on patches/server/0747-MC-Dev-fixes.patch:
```diff
@@ -0,0 +1,32 @@
+ // CraftBukkit start
+ import net.minecraft.SharedConstants;
+ 
+-public class Main {
```
Oops

NoahvdAa commented on patches/api/0329-Per-player-viewdistances.patch:
```diff
@@ -0,0 +1,112 @@
+      */
+     void setNoTickViewDistance(int viewDistance);
++
++    // Paper start - add view distances
```
This already seems to be in a Paper block.
```suggestion

```

NoahvdAa commented on patches/api/0329-Per-player-viewdistances.patch:
```diff
@@ -0,0 +1,112 @@
++     * @param viewDistance view distance in [2, 32] or -1
++     */
++    public void setSendViewDistance(int viewDistance);
++    // Paper end - add view distances
```
Same for this
```suggestion

```

Proximyst commented on patches/server/0748-Util-patch.patch:
```diff
@@ -0,0 +1,2999 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Spottedleaf <Spottedleaf@users.noreply.github.com>
+Date: Wed, 16 Jun 2021 19:32:15 -0700
```
Should perhaps be fixed up to the mcutils patch?

NoahvdAa commented on patches/api/0329-Per-player-viewdistances.patch:
```diff
@@ -0,0 +1,112 @@
++     */
++    public void setSendViewDistance(int viewDistance);
++    // Paper end - add view distances api
+     // Paper end
```
```suggestion

```

Proximyst reviewed (commented):
Small comments before my phone started slugging hard while viewing this lol

NoahvdAa commented on patches/server/0747-MC-Dev-fixes.patch:
```diff
@@ -0,0 +1,32 @@
+ import net.minecraft.SharedConstants;
+ 
+-public class Main {
++public class Main { // 
```
Leftover?
```suggestion

```

Owen1212055 commented on patches/server/0809-Boats-do-not-weigh-that-much.patch:
```diff
@@ -0,0 +1,33 @@
+                 entityboat.setYRot(user.getYRot());
+-                if (!world.noCollision(entityboat, entityboat.getBoundingBox().inflate(-net.minecraft.Util.COLLISION_EPSILON))) { // Paper
++                if (!world.noCollision(entityboat, entityboat.getBoundingBox())) { // Paper
+                     return InteractionResultHolder.fail(itemstack);
```
Are patches like these going to be merged into the paper equivalents? As it seems this just modifies a paper patch.
https://github.com/PaperMC/Paper/blob/8d182e8639a7d13d4ba8cade2c1532e0f0ef678c/patches/server/0733-Improve-boat-collision-performance.patch

NoahvdAa commented on patches/server/0749-Fix-incorrect-isRealPlayer-init.patch:
```diff
@@ -0,0 +1,31 @@
+     abstract public void loadAndSaveFiles(); // Paper - moved from DedicatedPlayerList constructor
+ 
+     public void placeNewPlayer(Connection connection, ServerPlayer player) {
++        player.isRealPlayer = true; // Paper // Paper - this is a better place to write this that works and isn't overriden by plugins
```
```suggestion
+        player.isRealPlayer = true; // Paper - this is a better place to write this that works and isn't overriden by plugins
```

kennytv replied:
Paper Paper, Ctrl+Shift+F after you've merged the patches (since it'll be in entirely new lines like here as well)

NoahvdAa commented on patches/server/0752-Not-implemeneted.patch:
```diff
@@ -0,0 +1,69 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Spottedleaf <Spottedleaf@users.noreply.github.com>
+Date: Sun, 3 Mar 2019 20:53:18 -0800
+Subject: [PATCH] Not implemeneted
```
```suggestion
Subject: [PATCH] Not implemented
```

kennytv commented on patches/server/0784-Add-packet-limiter-config.patch:
```diff
@@ -0,0 +1,205 @@
++            }
++
++            if (packetClazz == null || !net.minecraft.network.protocol.Packet.class.isAssignableFrom(packetClazz)) {
++                MinecraftServer.LOGGER.warn("Packet '" + packetClassName + "' does not exist, cannot limit it! Please update tuinity.yml");
```
Ctrl+Shift+F Tuinity for a couple more

NoahvdAa commented on patches/server/0760-Make-CallbackExecutor-strict-again.patch:
```diff
@@ -0,0 +1,85 @@
+-        // it's possible to schedule multiple tasks currently, so it's vital we change this impl
+-        // If we recurse into the executor again, we will append to another queue, ensuring task order consistency
+-        private java.util.Queue<Runnable> queue = new java.util.ArrayDeque<>(); // Paper - remove final
++        // Paper start - revert paper's change
```
Removing the old paper patch is probably better
```suggestion
+        // Paper start - revert paper's change
```

NoahvdAa commented on patches/server/0769-Do-not-run-close-logic-for-inventories-on-chunk-unlo.patch:
```diff
@@ -0,0 +1,68 @@
++    public void closeUnloadedInventory(org.bukkit.event.inventory.InventoryCloseEvent.Reason reason) {
++        // copied from above
++        CraftEventFactory.handleInventoryCloseEvent(this, reason); // CraftBukkit
++        // Paper end
```
Two paper ends?
```suggestion

```

NoahvdAa commented on patches/server/0785-Lag-compensate-block-breaking.patch:
```diff
@@ -0,0 +1,155 @@
+     private GameType previousGameModeForPlayer;
+     private boolean isDestroyingBlock;
+-    private int destroyProgressStart;
++    private int destroyProgressStart; private long lastDigTime; // Paper - lag compensate block breaking
```
```suggestion
+    private int destroyProgressStart;  // Paper - lag compensate block breaking
+    private long lastDigTime; // Paper - lag compensate block breaking
```

NoahvdAa commented on patches/server/0785-Lag-compensate-block-breaking.patch:
```diff
@@ -0,0 +1,155 @@
+     private boolean hasDelayedDestroy;
+     private BlockPos delayedDestroyPos;
+-    private int delayedTickStart;
++    private int delayedTickStart; private long hasDestroyedTooFastStartTime; // Paper - lag compensate block breaking
```
```suggestion
+    private int delayedTickStart; // Paper - lag compensate block breaking
+    private long hasDestroyedTooFastStartTime; // Paper - lag compensate block breaking
```

NoahvdAa commented on patches/server/0785-Lag-compensate-block-breaking.patch:
```diff
@@ -0,0 +1,155 @@
+                 }
+ 
+-                this.destroyProgressStart = this.gameTicks;
++                this.destroyProgressStart = this.gameTicks; this.lastDigTime = System.nanoTime(); // Paper - lag compensate block breaking
```
Why is this on the same line?

NoahvdAa commented on patches/server/0785-Lag-compensate-block-breaking.patch:
```diff
@@ -0,0 +1,155 @@
+                             this.hasDelayedDestroy = true;
+                             this.delayedDestroyPos = pos;
+-                            this.delayedTickStart = this.destroyProgressStart;
++                            this.delayedTickStart = this.destroyProgressStart; this.hasDestroyedTooFastStartTime = this.lastDigTime; // Paper - lag compensate block breaking
```
Same here, why is this on the same line?

NoahvdAa commented on patches/server/0787-Use-hash-table-for-maintaing-changed-block-set.patch:
```diff
@@ -0,0 +1,30 @@
+ import net.minecraft.server.MinecraftServer;
+ // CraftBukkit end
+ 
++import it.unimi.dsi.fastutil.shorts.ShortOpenHashSet; // Paper
```
Since this import is only used once, wouldn't a FQN be better here?

NoahvdAa commented on patches/server/0793-Make-entity-tracker-use-highest-range-of-passengers.patch:
```diff
@@ -0,0 +1,65 @@
+     static final org.spigotmc.TrackingRange.TrackingRangeType[] TRACKING_RANGE_TYPES = org.spigotmc.TrackingRange.TrackingRangeType.values();
+     public final com.destroystokyo.paper.util.misc.PlayerAreaMap[] playerEntityTrackerTrackMaps;
+-    final int[] entityTrackerTrackRanges;
++    final int[] entityTrackerTrackRanges; public int getEntityTrackerRange(final int ordinal) { return this.entityTrackerTrackRanges[ordinal]; } // Paper - public read
```
Is this some sort of Tuinity code style thing? Why is everything on the same line?

NoahvdAa commented on patches/server/0794-Oprimise-map-impl-for-tracked-players.patch:
```diff
@@ -0,0 +1,29 @@
+ import org.apache.logging.log4j.Logger;
+ 
+ import org.bukkit.entity.Player; // CraftBukkit
++import it.unimi.dsi.fastutil.objects.ReferenceOpenHashSet; // Paper
```
Import is used once, FQN would probably be better here.

HexedHero commented on patches/server/0766-Revert-MC-4-fix.patch:
```diff
@@ -0,0 +1,27 @@
+         // Paper start - fix MC-4
+         if (this instanceof ItemEntity) {
+-            if (com.destroystokyo.paper.PaperConfig.fixEntityPositionDesync) {
++            if (false && com.destroystokyo.paper.PaperConfig.fixEntityPositionDesync) { // Paper - revert
```
Probably don't want to add this patch since it just reverts another Paper patch?

kennytv commented on patches/server/0799-Optimise-CraftChunk-getEntities.patch:
```diff
@@ -0,0 +1,78 @@
+-            .filter(entity -> entity != null && entity.isValid())
+-            .toArray(Entity[]::new);
+-        // Paper end
++        return ((CraftWorld)this.getWorld()).getHandle().getChunkEntities(this.x, this.z); // Paper - optimise this better than paper :)
```
🦧

NoahvdAa reviewed (changes requested):
I scrolled through this very quickly and here are a few things I've noticed:

kennytv reviewed (commented):
Don't forget to run `./gradlew wrapper --gradle-version=7.2`

Proximyst commented on patches/api/0329-Per-player-viewdistances.patch:
```diff
@@ -0,0 +1,112 @@
+      *
+      * @return the player's view distance
+-     * @deprecated This is unimplemented and <i>will</i> throw an exception at runtime. The {@link org.bukkit.World World}-based methods still work.
++     * // Paper - implemented
```
This must be replaced with a HTML comment, if at all replaced. Perhaps it would be nice to add a comment to the general comment instead that this is in fact implemented on Paper, or `@since implemented in Paper 1.17.1`?
![image](https://user-images.githubusercontent.com/19861299/129848165-d022760b-abb9-4d51-981c-7b6a509c3ace.png)

Proximyst commented on patches/server/0748-Util-patch.patch:
```diff
@@ -0,0 +1,2999 @@
+new file mode 100644
+index 0000000000000000000000000000000000000000..be668387f65a633c6ac497fca632a4767a1bf3a2
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/util/CachedLists.java
```
Drop this file

Proximyst commented on patches/server/0748-Util-patch.patch:
```diff
@@ -0,0 +1,2999 @@
++
++    public static boolean isPowerOfTwo(final int n) {
++        // 2^n has one bit
++        // note: this rets true for 0 still
```
Is this deemed a bug?

Proximyst commented on patches/server/0748-Util-patch.patch:
```diff
@@ -0,0 +1,2999 @@
++public class Main { //
+ 
+     private static final Logger LOGGER = LogManager.getLogger();
+ 
```
Drop this hunk

Proximyst commented on patches/server/0748-Util-patch.patch:
```diff
@@ -0,0 +1,2999 @@
+     }
+ 
+-    private long chunkFutureAwaitCounter;
++    long chunkFutureAwaitCounter; // Paper - private -> package private
```
Access transformer

Proximyst commented on patches/server/0748-Util-patch.patch:
```diff
@@ -0,0 +1,2999 @@
+     // Paper end - rewrite ticklistserver
++    // Paper start
++    public final boolean areChunksLoadedForMove(AABB axisalignedbb) {
++        // copied code from collision methods, so that we can guarantee that they wont load chunks (we don't override
```
Make a comment for diff on change

Proximyst commented on patches/server/0750-Optimise-general-POI-access.patch:
```diff
@@ -0,0 +1,992 @@
+ 
+     @Nullable
+-    protected Optional<R> get(long pos) {
++    public Optional<R> get(long pos) { // Paper - public
```
A-T

Proximyst commented on patches/server/0750-Optimise-general-POI-access.patch:
```diff
@@ -0,0 +1,992 @@
+     }
+ 
+-    protected Optional<R> getOrLoad(long pos) {
++    public Optional<R> getOrLoad(long pos) { // Paper - public
```
A-T

Proximyst commented on patches/server/0751-Allow-controlled-flushing-for-network-manager.patch:
```diff
@@ -0,0 +1,147 @@
++        // Paper start - add flush parameter
++        this.a(packet, genericfuturelistener, enumprotocol, enumprotocol1, true);
++    }
++    private void a(Packet<?> packet, @Nullable GenericFutureListener<? extends Future<? super Void>> genericfuturelistener, ConnectionProtocol enumprotocol, ConnectionProtocol enumprotocol1, boolean flush) {
```
Let's consider giving this a proper name.

DenWav replied:
Based off of the `a` method above, which looks to not be correctly mapped. Best to figure out what the method name should be and go with that, and hopefully in time these mapping issues will happen less frequently.

Spottedleaf replied:
I'm marking this as resolved even though it's not, the concern here is completely valid but it's not something that should be addressed by this PR. if someone can commit the necessary fix to master, i can just pull it into this branch

Proximyst commented on patches/server/0754-Rewrite-the-light-engine.patch:
```diff
@@ -0,0 +1,4785 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Spottedleaf <spottedleaf@spottedleaf.dev>
+Date: Wed, 28 Oct 2020 16:51:55 -0700
+Subject: [PATCH] Rewrite the light engine
```
Should we consider dropping the other lighting engine changes we do for the sake of only keeping one patch set around for lighting engine optimisations?

Proximyst commented on patches/server/0754-Rewrite-the-light-engine.patch:
```diff
@@ -0,0 +1,4785 @@
+new file mode 100644
+index 0000000000000000000000000000000000000000..9efbdba758aebcad3454a9a52c8a7eae4b7fc7eb
+--- /dev/null
++++ b/src/main/java/ca/spottedleaf/starlight/light/BlockStarLightEngine.java
```
Is this package correct? I see you use spottedleaf.dev for your email now.

Proximyst commented on patches/server/0754-Rewrite-the-light-engine.patch:
```diff
@@ -0,0 +1,4785 @@
++        }
++
++        return ret;
++    }
```
Boxed booleans seems unnecessary here as they're all set to a binary of true and false, never `null`

Proximyst commented on patches/server/0754-Rewrite-the-light-engine.patch:
```diff
@@ -0,0 +1,4785 @@
+     private volatile CompletableFuture<Either<LevelChunk, ChunkHolder.ChunkLoadingFailure>> tickingChunkFuture; private volatile boolean isTickingReady; // Paper - cache chunk ticking stage
+     private volatile CompletableFuture<Either<LevelChunk, ChunkHolder.ChunkLoadingFailure>> entityTickingChunkFuture; private volatile boolean isEntityTickingReady; // Paper - cache chunk ticking stage
+-    private CompletableFuture<ChunkAccess> chunkToSave;
++    public CompletableFuture<ChunkAccess> chunkToSave; // Paper - public
```
A-T

Proximyst commented on patches/server/0754-Rewrite-the-light-engine.patch:
```diff
@@ -0,0 +1,4785 @@
+     public final ServerLevel level;
+     private final ThreadedLevelLightEngine lightEngine;
+-    private final BlockableEventLoop<Runnable> mainThreadExecutor;
++    public final BlockableEventLoop<Runnable> mainThreadExecutor; // Paper - public
```
A-T

Proximyst commented on patches/server/0754-Rewrite-the-light-engine.patch:
```diff
@@ -0,0 +1,4785 @@
+     private short tickingBlockCount;
+     private short tickingFluidCount;
+-    final PalettedContainer<BlockState> states; // Paper - package-private
++    public final PalettedContainer<BlockState> states; // Paper - package-private // Paper - public
```
A-T (consider adding a commented A-T for the original introduction of such a change)

Proximyst commented on patches/server/0754-Rewrite-the-light-engine.patch:
```diff
@@ -0,0 +1,4785 @@
+     }
+ 
+-    protected T get(int index) {
++    public T get(int index) { // Paper - public
```
A-T

Proximyst reviewed (commented):
Got around halfway down the page, will do the rest later today. Looking good so far :)
It goes without saying that both I and the rest of the community appreciates your work.

I'm not entirely sure around Paperweight's abilities with forks, but I assume there were no access transformers for you with that in Tuinity?

Spottedleaf committed:
Properly update gradlew

Spottedleaf committed:
Force clean rebuild

Spottedleaf committed:
Mark fixups

Comments and ATs still need to be done

Spottedleaf committed:
grep -r "Tuinity"

Spottedleaf committed:
Fixup

Spottedleaf committed:
Ensure gameprofile lastaccess is written only under the state lock

Spottedleaf committed:
update URL for dataconverter

Spottedleaf committed:
Only clean rebuild tuinity patches

might fix merge conflicts

Spottedleaf committed:
Use UTF-8 for gradlew

Spottedleaf committed:
Clean rb patches again

Spottedleaf committed:
Convert block ids used as item ids

Neither the converters of pre 1.13 nor DFU handled these cases,
as by the time they were written the game at the time didn't
consider these ids valid - they would be air. Because of this,
some worlds have logspam since only DataConverter (not DFU or
legacy converters) will warn when an invalid id has been
seen.

While quite a few do need to now be considered as air, quite a lot
do not. So it makes sense to add conversion for these items, instead
of simply suppressing or ignoring the logs. I've now added id -> string conversion
for all block ids that could be used as items that existed in the game
before 1.7.10 (I have no interest in tracking down the
exact version block ids stopped working) that were on
https://minecraft-ids.grahamedgecombe.com/

Items that did not directly convert to new items will
be instead converted to air: stems, wheat crops, piston head,
tripwire wire block

Spottedleaf committed:
Fix LightPopulated parsing in V1466

The DFU code was checking if the number existed, not if it
didn't exist. I misread the original code.

Spottedleaf committed:
Always parse protochunk light sources unless it is marked as non-lit

Chunks not marked as lit will always go through the light engine,
so they should always have their block sources parsed.

Spottedleaf committed:
Update custom names to JSON for players

Missed this fix from CB, as it was inside
the DataFixers class.

I decided to double check all of the CB changes again:

DataFixers.java was the only area I missed, as I had inspected all
datafixer diffs and implemented them all into DataConverter. I also
checked Bootstrap.java again, and re-evaluated their changes. I had
previously done this, but determined that they were all bad.

The changes to make standing_sign block map to oak_sign block in
V1450 is bad, because that's not the item id V1450 accepts. Only
in 1.14 did oak_sign even exist, and as expected there is a converter
to rename all existing sign items/blocks.

The fix to register the portal block under id 1440 is useless, as
the flattenning logic will default to the lowest registered id - which
is the exact blockstate that CB registers into 1440. So it just
doesn't do anything.

The extra item ids in the id -> string converter are already added,
but I found this from EMC originally.

The change for the spawn egg id 23 -> Arrow is just wrong,
that id DOES correspond to TippedArrow, NOT Arrow. As
expected, the spawn egg already has a dedicated mapping for
Arrow, which is id 10 - which was Arrow's entity id.

I also ported a fix for the cooked_fished id update. This doesn't
really matter since there is already a dataconverter to fix this,
but the game didn't accept cooked_fished at the time. So I see
no harm.

Spottedleaf committed:
Review all converters and walkers

- Refactor V99 to have helper methods for defining entity/tile
  entity types
- Automatically namespace all ids that should be namespaced.
  While vanilla never saved non-namespaced data for things that
  are namespaced, plugins/users might have.
- Synchronised the identity ensure map in HelperBlockFlatteningV1450
- Code style consistency
- Add missing log warning in V102 for ITEM_NAME type conversion
- Use getBoolean instead of getByte
- Use ConverterAbstractEntityRename for V143 TippedArrow -> Arrow
  rename, as it will affect ENTITY_NAME type
- Always set isVillager to false in V502 for Zombie
- Register V808's converter under subversion 1 like DFU
- Register a breakpoint for V1.17.1. In the future, all final
  versions of major releases will have a breakpoint so that
  the work required to determine if a converter needs a breakpoint
  is minimal
- Validate that a dataconverter is only registered for a version
  that is registered
- ConverterFlattenTileEntity is actually ConverterFlattenEntity
  It even registered the converters under TILE_ENTITY, instead of
  ENTITY.
- Fix id comparison in V1492 STRUCTURE_FEATURE renamer
- Use ConverterAbstractStatsRename for V1510 stats renamer
  At the time I had written that class, the abstract renamer didn't
  exist.
- Ensure OwnerUUID is at least set to empty string in
  V1904 if the ocelot is converted to a cat (this is
  likely so that it retains a collar)
- Use generic read/write for Records in V1946
  Records is actually a list, not a map. So reading map was
  invalid.

Spottedleaf committed:
Always set light to zero when propagating decrease

This fixes an almost infinite loop where light values
would be spam queued on a very small subset on blocks.

This also likely fixes the memory issues people were
seeing.

Spottedleaf committed:
Merge remote-tracking branch 'upstream/master' into merge-tuinity

Spottedleaf committed:
re-organize patches

Spottedleaf committed:
Apply and fix conflicts

Spottedleaf committed:
Revert some patches

getChunkAt retains chunks so that plugins don't spam loads
revert mc-4 fix will remain unless issues pop up

Spottedleaf committed:
Shuffle iterated chunks if per player is not enabled

Can help with some mob spawning stacking up at locations

Spottedleaf committed:
Make per player default, migrate all configs

Spottedleaf committed:
Adjust comments in fixups

Spottedleaf committed:
Rework config for player chunk loader

Old config is not compatible. Move all configs to be
under `settings` in paper.yml

The player chunk loader has been modified to
less aggressively load chunks, but to send
chunks at higher rates compared to tuinity. There are
new config entries to tune this behavior.

jpenilla committed:
Add back old constructor to CompressionEncoder/Decoder (fixes
 Tuinity #358)

Spottedleaf committed:
Raise chunk loading default limits

Spottedleaf committed:
Reduce worldgen thread workers for lower core count cpus

Spottedleaf committed:
Raise limits for chunk loading config

Also place it under `chunk-loading`

Spottedleaf committed:
Disable max chunk send rate by default

Spottedleaf committed:
Merge remote-tracking branch 'upstream/master' into merge-tuinity

Spottedleaf committed:
Fix conflicts and rebuild patches

Spottedleaf committed:
Drop default send rate again

Appears to be still causing problems for no known reason

electronicboy reviewed (approved):
I've looked over this a few times, and bar there being too much to go line by line nit-picking, the patches here seem good overall, the work in here has already heavily been battle-tested so I personally have no reasons to hold this up, so, giving my approval

probably some random bits and bats like ATs which could be cleaned up but given the scope here that sounds like busy-work that can be done over time/as needed

Spottedleaf committed:
Raise chunk send limits to 100 per player

While a low limit fixes ping issues for some people, most people
do not suffer from this issue and thus should not suffer from
an extremely slow load-in rate.

Spottedleaf committed:
Rebase part 1

Autosquash the fixups

Spottedleaf committed:
Move not implemented up

Spottedleaf committed:
Fixup mc-dev fixes

Missed this one

Spottedleaf committed:
Rebase per player viewdistance api into the original api patch

Spottedleaf committed:
Remove old light engine patch part 1

The prioritisation must be kept from it, so that part
has been rebased into the priority patch.
Part 2 will deal with rebasing all of the patches _after_

Spottedleaf committed:
Rebase remaining patches for old light patch removal

Spottedleaf committed:
Remove other mid tick patch

Spottedleaf committed:
Remove Optimize-PlayerChunkMap-memory-use-for-visibleChunks.patch

Replaced by `Do not copy visible chunks`

Spottedleaf committed:
Revert AT for Vec3i setX/Y/Z

The class is immutable. set should not be exposed

Spottedleaf committed:
Remove old IntegerUtil class

Spottedleaf committed:
Replace old CraftChunk#getEntities patch

Spottedleaf committed:
Remove import for SWMRNibbleArray in ChunkAccess

Spottedleaf committed:
Finished merge checklist

Spottedleaf committed:
Remove ensureTickThread impl in urgency patch
